spec/inputs/jdbc_spec.rb

```diff
@@ -260,4 +260,34 @@
       expect(File).not_to exist(settings["last_run_metadata_path"])
     end
   end
+
+  context "when setting fetch size" do
+
+    let(:settings) do
+      {
+        "statement" => "SELECT * from test_table",
+        "jdbc_fetch_size" => 1
+      }
+    end
+
+    let(:num_rows) { 10 }
+
+    before do
+      num_rows.times do
+        db[:test_table].insert(:num => 1, :created_at => Time.now.utc)
+      end
+
+      plugin.register
+    end
+
+    after do
+      plugin.teardown
+    end
+
+    it "should fetch all rows" do
```

Comment:
I am mainly testing that the library doesn't throw an exception when this parameter is set. fetch_size is defaulted to `10` in some clients. this is not how much of the result set is returned... but how much is cached before the cursor is moved and another batch is fetched from the database. It is more for keeping memory footprint low while trying to retrieve a _really_ large result.